src/main/java/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -1490,24 +1493,23 @@ private long uncheckedReadFlexInt_1_1() {
      * Marker's endIndex is set to the symbol ID value and its startIndex is set to -1. When this FlexSym wraps a
      * delimited end marker, neither the Marker's startIndex nor its endIndex is set.
      * @param markerToSet the marker to populate.
-     * @return the symbol ID value if one was present, otherwise -1.
+     * @return the user-space symbol ID value if one was present, otherwise -1.
      */
     private long uncheckedReadFlexSym_1_1(Marker markerToSet) {
         long result = uncheckedReadFlexInt_1_1();
         if (result == 0) {
             int nextByte = buffer[(int)(peekIndex++)];
-            if (nextByte == OpCodes.INLINE_SYMBOL_ZERO_LENGTH) {
+            // TODO: We could pretend $0 is a system symbol and consolidate some of the branches here. Is it worth it?
+            if (nextByte == FLEX_SYM_SYSTEM_SYMBOL_OFFSET) {
```

Comment:
I'd be fine with that, if you can get it to work. However, we return 0 from this branch and -1 from the system symbol branch, so there may need to be the same amount of branching either way.